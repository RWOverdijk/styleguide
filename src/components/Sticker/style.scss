/** @define Sticker */

@import "../../css/variables";

.Van-Sticker {
  position: relative;
  z-index: 10;
  width: 106px;
  height: 106px;
  color: $color-white;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;

  /* postcss-bem-linter: ignore */
  > span,
  > div {
    line-height: 1.1;
    z-index: 2;
  }

  &--circle {
    background-color: $color-blue-primary;
    border-radius: 50%;
  }

  &--splash {
    transform: rotate(10deg);

    /* postcss-bem-linter: ignore */
    > span:not(.Van-Sticker-icon),
    > div {
      width: 90px;
    }
  }

  &-icon {
    position: absolute;
    width: 100px;
    height: auto;
    z-index: 1;
  }

  @media (max-width: $container-sm) {
    width: 100px;
    height: 100px;

    &--splash {
      /* postcss-bem-linter: ignore */
      > span:not(.Van-Sticker-icon),
      > div {
        width: 50px;
      }
    }

    &-icon {
      width: 90px;
    }
  }
}
