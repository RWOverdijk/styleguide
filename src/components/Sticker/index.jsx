import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';
import sunIcon from '../../assets/icons/ftic/sun.svg';
import './style.scss';

class Sticker extends PureComponent {
  render() {
    const { className, children, type, ...props } = this.props;

    return (
      <span
        className={classnames(
          'Van-Sticker',
          {
            'Van-Sticker--circle': type === 'circle',
            'Van-Sticker--splash': type === 'splash'
          },
          className
        )}
        {...props}
      >
        {type === 'splash' && (
          <Icon source={sunIcon} className="Van-Sticker-icon Van-u-color-yellow-primary" />
        )}
        {typeof children === 'string' ? <span>{children}</span> : children}
      </span>
    );
  }
}

Sticker.defaultProps = {
  className: '',
  children: '',
  type: 'circle'
};

Sticker.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  type: PropTypes.oneOf(['circle', 'splash'])
};

export default Sticker;
