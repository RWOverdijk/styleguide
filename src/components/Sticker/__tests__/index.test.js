import React from 'react';
import renderer from 'react-test-renderer';
import Sticker from '../';

describe("<Sticker>", () => {
  it("renders default Sticker", () => {
    const tree = renderer.create(
      <Sticker>My sticker</Sticker>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders splash Sticker", () => {
    const tree = renderer.create(
      <Sticker type="splash">My sticker</Sticker>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Sticker className="my-class">My sticker</Sticker>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Sticker data-id="0">My sticker</Sticker>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
