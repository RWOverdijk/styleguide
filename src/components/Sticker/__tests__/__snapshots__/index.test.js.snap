// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sticker> renders default Sticker 1`] = `
<span
  className="Van-Sticker"
>
  My sticker
</span>
`;

exports[`<Sticker> renders with specific attributes 1`] = `
<span
  className="Van-Sticker"
  data-id="0"
>
  My sticker
</span>
`;

exports[`<Sticker> renders with specific className 1`] = `
<span
  className="Van-Sticker my-class"
>
  My sticker
</span>
`;
